src/CamrecScreenStreamDecoder/CamrecScreenStreamDecoder.cpp

```diff
@@ -0,0 +1,270 @@
+      for ( uint32_t i = 0; i < frameCount; ++i )
+      {
+         uint32_t frameSize = avi20Bundle->mediaStreamReader->FrameSize( i );
+         if ( frameSize > compressedSize )
+         {
+            pCompressed.reset( new uint8_t[frameSize] );
+            compressedSize = frameSize;
+         }
+
+         bool status = avi20Bundle->mediaStreamReader->ReadFrameData( i, pCompressed.get() );
+         if ( !status )
+            break;
+
+         bmihCompressed.biSizeImage = frameSize;
+         DWORD flags = avi20Bundle->mediaStreamReader->IsKeyframe( i ) ? 0 : ICDECOMPRESS_NOTKEYFRAME;
+         DWORD decompressStatus = ::ICDecompress( hic, flags, &bmihCompressed, pCompressed.get(), &bmihDecompresed, pUncompressed.get() );
+         if ( decompressStatus != ICERR_OK )
+            break;
+
+         if ( i == 0 )
+         {
+            int stride = bmihDecompresed.biWidth * bmihDecompresed.biBitCount / 8;
+            PixelFormat pf = ( bmihDecompresed.biBitCount == 24 ) ? PixelFormat24bppRGB : PixelFormat32bppRGB;
+            BYTE *pLastRow = pUncompressed.get() + stride * (bmihDecompresed.biHeight - 1);
+            Gdiplus::Bitmap bm( bmihDecompresed.biWidth, bmihDecompresed.biHeight, -stride, pf, pLastRow );
+
+            CAtlString outputPath;
+            TCHAR buf[MAX_PATH];
+            ::GetTempPath( MAX_PATH, buf );
+            outputPath.Format( L"%s\\frame_%04u.png", buf, i );
+            QuickSavePic( &bm, outputPath );
+         }
+ 
+         ++numDecoded;
+      }
+
+      if ( hic )
+      {
+         ::ICDecompressEnd( hic );
+         ::ICClose( hic );
+      }
+
+      avi20Bundle.reset();
+      pStream.Release();
+      pStorage.Release();
+
+      return numDecoded == frameCount;
+   }
+}
+
+
+int main()
+{
+   std::string camrecWithTypeTwoAVI( "C:\\git\\CamtasiaWin3\\TSCTestMediaFiles\\1.4 part 4 remake.camrec" );
+   //std::string camrecWithTypeOneAVI( "C:\\crap\\Tech Support\\Project\\OL RT 1 with math  & changes.pptx1-5.camrec" );
+   std::string camrecWithTypeOneAVI( "C:\\Test Media\\camrec\\Customer Dictionary.camrec" );
+
+   HRESULT hr = ::CoInitializeEx( nullptr, COINIT_APARTMENTTHREADED );
+   if ( FAILED( hr ) )
+      return -1;
+
+   Gdiplus::GdiplusStartupInput gdiplusStartupInput;
+   Gdiplus::GdiplusStartupOutput gdiplusStartupOutput;
+   ULONG_PTR gdiplusToken, gdiplusBGThreadToken;
+   gdiplusStartupInput.SuppressBackgroundThread = TRUE;
+   Gdiplus::GdiplusStartup( &gdiplusToken,
+                            &gdiplusStartupInput, &gdiplusStartupOutput );
+   Gdiplus::Status stat = gdiplusStartupOutput.NotificationHook( &gdiplusBGThreadToken );
+
+   //bool status = DecodeCamrecScreenStream( camrecWithTypeTwoAVI );
+   bool status = DecodeCamrecScreenStream( camrecWithTypeOneAVI );
+   assert( status == true );
+
+   gdiplusStartupOutput.NotificationUnhook( gdiplusBGThreadToken );
+   Gdiplus::GdiplusShutdown( gdiplusToken );
+
+   ::CoUninitialize();
+    return 0;
+}
```

Comment:
did you intend to check this file in? seems like a test case for a local file

Reply:
That project has been updated to be two integration tests.

Reply:
And... I removed these integration tests entirely an instead added a golden-frame test.